yarn-project/end-to-end/src/e2e_deploy_contract.test.ts

```diff
@@ -273,121 +259,38 @@ describe('e2e_deploy_contract', () => {
       expect(registeredClass!.packedBytecode.toString('hex')).toEqual(contractClass.packedBytecode.toString('hex'));
       expect(registeredClass!.publicFunctions).toEqual(contractClass.publicFunctions);
       expect(registeredClass!.privateFunctions).toEqual([]);
     });
 
-    it('broadcasts a private function and registers it on the node', async () => {
-      const privateFunction = contractClass.privateFunctions[0];
-      const privateFunctionArtifact = artifact.functions.find(fn =>
-        FunctionSelector.fromNameAndParameters(fn).equals(privateFunction.selector),
-      )!;
-
-      // TODO(@spalladino): The following is computing the unconstrained root hash twice.
-      // Feels like we need a nicer API for returning a hash along with all its preimages,
-      // since it's common to provide all hash preimages to a function that verifies them.
-      const artifactMetadataHash = computeArtifactMetadataHash(artifact);
-      const unconstrainedArtifactFunctionTreeRoot = computeArtifactFunctionTreeRoot(artifact, FunctionType.OPEN);
-
-      // We need two sibling paths because private function information is split across two trees:
-      // The "private function tree" captures the selectors and verification keys, and is used in the kernel circuit for verifying the proof generated by the app circuit.
-      // The "artifact tree" captures function bytecode and metadata, and is used by the pxe to check that its executing the code it's supposed to be executing, but it never goes into circuits.
-      const privateFunctionTreePath = computePrivateFunctionsTree(contractClass.privateFunctions).getSiblingPath(0);
-      const artifactFunctionTreePath = computeArtifactFunctionTree(artifact, FunctionType.SECRET)!.getSiblingPath(0);
-
-      const selector = privateFunction.selector;
-      const metadataHash = computeFunctionArtifactHash(privateFunctionArtifact);
-      const bytecode = bufferAsFields(
-        Buffer.from(privateFunctionArtifact.bytecode, 'hex'),
-        MAX_PACKED_BYTECODE_SIZE_PER_PRIVATE_FUNCTION_IN_FIELDS,
-      );
-      const vkHash = privateFunction.vkHash;
-
-      await registerer.methods
-        .broadcast_private_function(
-          contractClass.id,
-          Fr.fromBufferReduce(artifactMetadataHash),
-          Fr.fromBufferReduce(unconstrainedArtifactFunctionTreeRoot),
-          privateFunctionTreePath.map(Fr.fromBufferReduce),
-          padArrayEnd(artifactFunctionTreePath.map(Fr.fromBufferReduce), Fr.ZERO, ARTIFACT_FUNCTION_TREE_MAX_HEIGHT),
-          // eslint-disable-next-line camelcase
-          { selector, metadata_hash: Fr.fromBufferReduce(metadataHash), bytecode, vk_hash: vkHash },
-        )
-        .send()
-        .wait();
+    it('broadcasts a private function', async () => {
+      const selector = contractClass.privateFunctions[0].selector;
+      await broadcastPrivateFunction(wallet, artifact, selector).send().wait();
```

Comment:
So I understand, the test will eventually show that another PXE can point to an instance of this class, and use this private function without having the full contract class artifact?